tools/jit/gen_jit_dispatch.py

```diff
@@ -254,8 +271,7 @@ def declkey(decl):
         arguments = decl['arguments']
         for n, arg in enumerate(arguments):
             if arg['simple_type'] == 'TensorOptions':
-                del arguments[n]
-                arguments.extend([
+                decl['arguments'] = arguments[:n] + [
```

Comment:
This can be more easily understood as a flat map:
```
def  expand_options(i, a):
  if a['simple_type'] != 'TensorOptions':
    return [a]
  assert('tensor_options_arg_index' not in n)
  decl['tensor_options_arg_index'] = n
  return [ {...}, {...}, {...} ]
decl['arguments'] = [ a for x in expand_options(i, arg) for i, arg in enumerate(decl['arguments']) ]
```

Reply:
Thanks for the suggestion! 

Is tracing disabled on methods with `TensorOptions` argument? `.to(device)` now dispatches to `at::Tensor::to(TensorOptions)` and it seems to be traced as a no-op. I haven't looked at the JIT generated code, but will do tomorrow.

Reply:
Tracing should be enabled. We trace torch.zeros, for instance. Not sure why `to` doesn't trace then.

Reply:
Alright then. I'll investigate tomorrow!